niworkflows/interfaces/tests/test_bids.py

```diff
@@ -382,6 +382,55 @@ def test_DerivativesDataSink_t1w(tmp_path, space, size, units, xcodes, fixed):
     assert nii.header.get_xyzt_units() == ("mm", "unknown")
 
 
+@pytest.mark.parametrize(
+    "source_file",
+    [
+        BOLD_PATH,
+        [BOLD_PATH],
+        [BOLD_PATH, "ds054/sub-100185/func/sub-100185_task-machinegame_run-02_bold.nii.gz"]
+    ]
+)
+@pytest.mark.parametrize("source_dtype", ["<i4", "<f4"])
+@pytest.mark.parametrize("in_dtype", ["<i4", "<f4"])
+def test_DerivativesDataSink_data_dtype_source(
+    tmp_path, source_file, source_dtype, in_dtype
+):
+
+    def make_empty_nii_with_dtype(fname, dtype):
+        Path(fname).parent.mkdir(exist_ok=True, parents=True)
+
+        size = (30, 30, 30, 10)
```

Comment:
No real need to spend a megabyte on this.

```suggestion
        size = (2, 3, 4, 5)
```